### PR TITLE
Hcap 173 employer site entity

### PR DESCRIPTION
**Pull request by mauriciofreshw**
Refactored @fwpushan branch with the new requirements

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -45,13 +45,24 @@ local-close:
 	@echo "Stopping local app container"
 	@docker-compose -f docker-compose.dev.yml down
 
+local-clean:
```
Comment (fwkendall) [inline Makefile:48]: Is this in here twice now...?

File: server/tests/employer.form.test.js [hunk 2]
```diff
@@ -17,6 +19,25 @@ describe('Server V1 Form Endpoints', () => {
 
   const formEndpoint = '/api/v1/employer-form';
 
+  const site = {
+    siteId: 67,
+    siteName: 'Test site',
+    earlyAdaptorAllocation: 1,
```
Comment (fwkendall) [inline server/tests/employer.form.test.js:25]: minor typo - this should be adopter, with an o :)

File: server/tests/employer.form.test.js [hunk 3]
```diff
@@ -150,4 +171,35 @@ describe('Server V1 Form Endpoints', () => {
...
+
+  it('Create new site, receive success', async () => {
+    const res = await saveSites(site);
+    const expectedRes = [
+      { siteId: 67, status: 'Success' },
+    ];
+    expect(res).toEqual(expectedRes);
+  });
+
+  it('Get sites, receive all successfully', async () => {
```
Comment (fwkendall) [inline server/tests/employer.form.test.js:183]: I'm not super familiar with this testing framework, do the tests always run in order?

File: server/validation.js [hunk 4]
```diff
@@ -154,31 +155,56 @@ const EmployerFormSchema = yup.object().noUnknown('Unknown field in form').shape
   doesCertify: yup.boolean().typeError(errorMessage).required(errorMessage).test('is-true', errorMessage, (v) => v === true),
 });
 
+const EmployerSiteBatchSchema = yup.array().of(
+  yup.lazy((item, options) => {
+    const index = options.parent.indexOf(item);
+    return yup.object().noUnknown(`Unknown field in site data (index ${index})`).shape({
```
Comment (fwkendall) [inline server/validation.js:161]: Just a note on some fields that I think are required and not nullable:

- siteId
- siteName
- healthAuthority
- postalCode

I think too many other parts of the application hinge on these fields to allow them to be nullable. We need the site name in order to show it in the access approval dropdown, and we need the HA + postal code for region access controls and sorting/searching later

**Comment by fwkendall** [comment]
Nice work, especially appreciate the test coverage! Just a few comments from me :)

File: server/tests/employer.form.test.js (hunk 3 above)
Comment (mauriciofreshw) [reply server/tests/employer.form.test.js:183]: Yes, we use `--runInBand` when running the test command

File: Makefile (hunk 1 above)
Comment (mauriciofreshw) [reply Makefile:48]: Good catch!

**Review by fwkendall — APPROVED** [review]

**Review by Zenen-freshworks — APPROVED** [review]
Looks good to me!

**Pull request merged**